onnx/backend/test/cmd_tools.py

```diff
@@ -45,18 +45,40 @@ def prepare_dir(path):  # type: (Text) -> None
                 data_set_dir = os.path.join(
                     output_dir, 'test_data_set_{}'.format(i))
                 prepare_dir(data_set_dir)
-                for j, input_np in enumerate(inputs):
-                    tensor = numpy_helper.from_array(
-                        input_np, case.model.graph.input[j].name)
+                for j, input in enumerate(inputs):
+                    if isinstance(input, dict):
```

Comment:
A minor suggestion: it may be more general to use "collections.abc.Mapping" instead of "dict". Similarly, "collections.abc.iterable" or "collections.abc.Sequence" might be better than "list" below. However, this would require us to move the test for ndarray up first, since an array might also be an iterable/sequence object. (I haven't tried it out, so don't know if anything will break, so do test it out if you try this. https://stackoverflow.com/questions/35690572/what-does-isinstance-with-a-dictionary-and-abc-mapping-from-collections-doing has more info on this. (I am not fully familiar with pythonic idioms.)

Reply:
After a quick test (shown below) -- I've updated the listings with the abc mappings accordingly. Highlight: lists are abc.Sequence but arrays are not.

>>> import numpy as np
>>> from collections import abc
>>> arr = np.array([1, 2, 3])
>>> lst = [4, 5, 6]
>>> isinstance(arr, abc.Sequence)
False
>>> isinstance(lst, abc.Sequence)
True
>>> isinstance(arr, np.ndarray)
True
>>> isinstance(arr, abc.Iterable)
True
>>> isinstance(lst, abc.Iterable)
True

Reply:
Ultimately decided to not use abc functionality as collections.abc is not included in Python 2.7